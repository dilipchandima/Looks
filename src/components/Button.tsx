import React from 'react';
import type {ReactElement} from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';

import {Colors} from 'res/Colors';

type Props = {
  onPress(): void;
  title: string;
  icon?: ReactElement;
};

const Button = ({onPress, title, icon}: Props) => {
  return (
    <TouchableOpacity style={styles.root} onPress={onPress}>
      {icon}
      <Text style={styles.title}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  root: {
    height: 50,
    paddingHorizontal: 20,
    flexDirection: 'row',
    alignItems: 'center',
  },
  title: {
    color: Colors.white,
    fontSize: 20,
    marginLeft: 10,
  },
});

export default Button;
