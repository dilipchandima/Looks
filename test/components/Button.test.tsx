import React from 'react';

import {render} from '@testing-library/react-native';
import Button from 'components/Button';

const onPress = jest.fn();

describe('ButtonComponent', () => {
  it('should match to snapshot', () => {
    const screen = render(<Button title="NEXT" onPress={onPress} />);
    expect(screen).toMatchSnapshot();
  });
});
