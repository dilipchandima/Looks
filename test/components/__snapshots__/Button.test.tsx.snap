// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent should match to snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 50,
      "opacity": 1,
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 20,
        "marginLeft": 10,
      }
    }
  >
    NEXT
  </Text>
</View>
`;
